File: ios/equipapp/equipapp/Table/Question.swift
//
//  Question.swift
//  equipapp
//
//  Created by Nathan Sinclair on 5/12/18.
//  Copyright © 2018 Nathan Sinclair. All rights reserved.
//

import UIKit

class Question: NSObject {
    
    var speaker: String
    var author: String
    var question: String
    var cr_date: Date
    
    init(newSpeaker: String, newAuthor: String, newQuestion: String, newDate: Date) {
        self.speaker = newSpeaker
        self.author = newAuthor
        self.question = newQuestion
        self.cr_date = newDate
    }

}

File: android/app/src/main/java/com/nathan/equipapp/assets/Question.kt
package com.nathan.equipapp.assets

import org.threeten.bp.DateTimeUtils
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneId
import java.util.*

class Question {

    var author: String
    var description: String
    var speaker: String
    var date: Date
    var formattedDate: String? = ""
    var documentID: String? = ""

            constructor (newSpeaker: String, newAuthor: String, newDescription: String, newDate: Date, id: String) {
                this.speaker = newSpeaker
                this.author = newAuthor
                this.description = newDescription
                this.date = newDate
                documentID = id
                createDate()
            }



    fun createDate() {

        var formatter = org.threeten.bp.format.DateTimeFormatter.ofPattern("eee, d MMMM YYYY")
        val aucklandTime = LocalDateTime.ofInstant(DateTimeUtils.toInstant(date), ZoneId.of("Pacific/Auckland"))
        formattedDate = aucklandTime.format(formatter)

    }
}